### ISSUE TITLE
Bug: Latest Rancher seems to be incompatible with v1.30.4 +rke2r1

### ISSUE THREAD

**Issue by Sacrifice4135** [rancher/dashboard#11780]
**Setup**
- Rancher version: v2.9.1
- Rancher UI Extensions: None
- Browser type & version: Mozilla Firefox 128.0.3 (64-bit)
- Kubernetes Version: v1.30.4 +rke2r1. 
- Rancher Type: Helm Chart install on Kubernetes Cluster

**Describe the bug**
1. Bug: When I try to adjust the Cluster Appearance, I get the Error Message: "cannot upgrade cluster version from [v1.30.4+rke2r1] to [v1.28.12+rke2r1]. New version must be higher."

2. Bug: When I start a Helm Install/Upgrade of a App it doesn't shows logs anymore. Example with a Gitlab Helm Chart. I just get the start message and a message when its finished:

```
helm upgrade --history-max=5 --install=true --labels=catalog.cattle.io/cluster-repo-name=gitlab --namespace=gitlab --timeout=15m0s --values=/home/shell/helm/values-gitlab-8.3.1.yaml --version=8.3.1 --wait=true gitlab /home/shell/helm/gitlab-8.3.1.tgz
Release "gitlab" has been upgraded. Happy Helming!
NAME: gitlab
LAST DEPLOYED: Fri Aug 30 09:41:28 2024
NAMESPACE: gitlab
STATUS: deployed
REVISION: 9
NOTES:
=== CRITICAL
The following charts are included for evaluation purposes only. They will not be supported by GitLab Support
for production workloads. Use Cloud Native Hybrid deployments for production. For more information visit
https://docs.gitlab.com/charts/installation/index.html#use-the-reference-architectures.
- Redis
- Gitaly
- MinIO
=== NOTICE
The minimum required version of PostgreSQL is now 14. See https://docs.gitlab.com/charts/installation/upgrade.html for more details.
=== NOTICE
You've installed GitLab Runner without the ability to use 'docker in docker'.
The GitLab Runner chart (gitlab/gitlab-runner) is deployed without the `privileged` flag by default for security purposes. This can be changed by setting `gitlab-runner.runners.privileged` to `true`. Before doing so, please read the GitLab Runner chart's documentation on why we
chose not to enable this by default. See https://docs.gitlab.com/runner/install/kubernetes.html#running-docker-in-docker-containers-with-gitlab-runners
---------------------------------------------------------------------
SUCCESS: helm upgrade --history-max=5 --install=true --labels=catalog.cattle.io/cluster-repo-name=gitlab --namespace=gitlab --timeout=15m0s --values=/home/shell/helm/values-gitlab-8.3.1.yaml --version=8.3.1 --wait=true gitlab /home/shell/helm/gitlab-8.3.1.tgz
---------------------------------------------------------------------
```

**To Reproduce**
1. Installed Rancher with the Helm Chart on a Kubernetes Cluster with RKE2 v1.28 rke2r1 in the channel "rancher-stable"
2. Installed the Rancher System Upgrade Controller, with is configured to install RKE2 out of the channel "stable".
3. The System Upgrade Controller started to upgrade all Nodes to Kubernetes v1.30.4 +rke2r1
4. I've got the warning, that Rancher is incompatible with v1.30.4 +rke2r1.
5. I've changed the repo channel from "rancher-stable" to "rancher-latest" and installed the latest version.
6. Since that I have this Bugs.

**Result**

**Expected Result**
- Normally I can adjust the cluster appereance.
- A Helm Install/Upgrade should show a detailed log of whats happening, like the older Rancher version did.

**Screenshots**
![image](https://github.com/user-attachments/assets/5773a7d9-86d7-4fdb-a0e3-30d055266bff)

**Comment by richard-cox** [comment]
@Sacrifice4135 Unfortunately I can't transfer this issue over to https://github.com/rancher/system-upgrade-controller. Can you recreate there and reference my comment above?

**Comment by Sacrifice4135** [comment]
> It looks like the system upgrader hasn't correctly updated all required resources.
> 
> Cluster appearance is stored on the `/v3/cluster` resource which looks like it has a higher version of kube than what's tracked as installed elsewhere.

I've installed Rancher at v1.28. Rancher also recognizes the new Version correctly:
![image](https://github.com/user-attachments/assets/47df74f2-2527-436f-bf5a-5d30dfb9ed88)

But in the Rancher Cluster-Config were an old version. Today I had the ability to adjust the Kubernetes Version in the Cluster Configuration:
![image](https://github.com/user-attachments/assets/69b959e5-10db-4c0d-83f3-51c8e0ade88e)

Last 2 Days there we're the Options v1.28, v1.30, v1.30.3 but not v1.30.4. But today, I dont know why, there spawned the option for v1.30.4 and I've upgraded the configuration. Since then the Issues are gone.
